services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/[asset].js

```diff
@@ -231,7 +224,7 @@ Asset.propTypes = {
   libraryData: libraryPropTypes
 }
 
-export const getStaticProps = async ({ params }) => {
+export const getServerSideProps = async ({ params }) => {
```

Comment:
Long term, we'll want this to be ISR for optimal performance with on-demand regeneration. For for now, SSR seems totally fine for me. Our file system cache is caching GitHub requests, so each subsequent page view should read from the file system, get content, return server side rendered page. Each page request shouldn't fire off a GitHub request for this, so all good.

Cool with this @jdharvey-ibm?

If updating this page to SSR, lets do the same for `services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/index.js`

Reply:
@mattrosno  Does nextjs provide a "deferred ISR" type of thing? As in: something that marks the pages as incrementally regenerated, but doesn't attempt to build them for the first time during the production build itself?

One of the problems we're hitting here is that at prod build time, we're slamming the github server with requests from the same personal access token to statically generate the initial copy of these pages. As the number of assets grows, it will become less and less feasible to do that and we're already basically at the limit of what a single token can do.

Also, once we have the data graph up and running, this data won't be available to the web-app at build time; it'll be maintained throughout the rest of the service infrastructure. When that happens, it won't be possible to obtain a comprehensive list of all components during the web app builds to populate `getStaticPaths`.

We can definitely keep a close eye on performance, but we'll likely have to use gSSP for most of these requests and rely on caching through the rest of the service infrastructure to keep things snappy.

Reply:
ISR doesn't build pages during the production build (`npm run build`.) Our issue is that the catalog pages are statically generated, and each of those catalog pages (and the libraries page) fetch all libraries (and their assets).

Reply:
Long term, we'll want library and detail pages ISR, and then catalog pages to be some hybrid where the catalog content is fetched either server side or client side when the page hydrates. So there's no dependency on GitHub data at production build time.

Reply:
now that I'm thinking about this, I think we could keep this as `getStaticProps` and get the same result; Matt's right these pages are not being built at build time. as long as we don't fetch the issueCount on`getLibraryData` we should be good.
TLDR: we can change `getServerSideProps` back to `getStaticProps` and re-add the `getStaticPaths` method

Reply:
What is the action item for this PR? 😄

Reply:
To clarify, are you and @jdharvey-ibm recommending the former `getStaticProps` in this page or the new `getServerSideProps`?

Reply:
Yeah I'm totally fine with adjusting any/all of the data fetching. As long as it's cached/reduces the number of requests that happen at build time, I'm good with whatever.